### DISCUSSION TITLE
The problem of non-uniform step size in FAST & Simulink joint simulation（Step size selection）

### DISCUSSION THREAD

**Discussion by Sunwannasleep** [OpenFAST/openfast#1378] [Q&A]
I built the grid-connected control model of wind power in simulink, as shown in Figure 1. I want to realize the joint simulation of FAST and Simulink, but I found that the step size is different. The FAST default step, which named DT, is 0.0125（in _5MW_OC4Semi_WSt_WavesWN.fst_ file）, and the simulation step size of Simulink model is 5e-6. 
As you can see, the simulation step size is inconsistent. I tried to change the step size DT of FAST to 5e-6, but I found that the operation process became very, very slow. So, I am wondering if you have suggestions to solve the problem.
![1670990279510](https://user-images.githubusercontent.com/111493977/207502338-e84e1b3c-2b83-4141-9b91-ee6e8a0f9233.png)

Best regards,

**Comment by jjonkman** [comment]
Dear @Sunwannasleep,

I'm not surprised that dropping the time step significantly in the FAST model would lead to very long simulation times.

I'm not familiar with the details of your Simulink model, but is there a reason why you need such a small time step in Simulink?

If this is because you are trying to resolve high-frequency dynamics within Simulink, I would say that you have a large mismatch in resolved frequencies between the FAST and Simulink models.  Two-way coupling is really only important if you are resolving similar a frequency content on both sides of the interface.  Is there way to resolve only the low frequencies in the Simulink model, consistent with the frequencies in FAST such that the Simulink model can use a larger time step consistent with the FAST model?

Best regards,

**Reply by Sunwannasleep** [reply]
Dear @jjonkman  ,
The powergui in Simulink selects the discrete type (with sample time is 5e-6). If I change the sample time to the FAST default value of 0.0125, the simulation will report an error(the error is 'there may be a singularity in the solution').
By the way, selecting continuous types in powergui can't implement this process well.

**Reply by jjonkman** [reply]
I don't have anything to add.

**Reply by Sunwannasleep** [reply]
Thanks a lot！
